src/utility/SuggestionsManager.ts

```diff
@@ -16,86 +18,162 @@ export abstract class SuggestionsManager {
+        let toJSON = JSON.stringify(objToStringify);
+        fs.writeFileSync(ActBotConstants.suggestionsPath, toJSON);
+    }
+
+    /**
+     * Converts the collection of list/suggestions into an object for strigifying.
+     */
+    public static collectionToObject() {
+        let obj = Object.create(null);
+        for (let [k, v] of SuggestionsManager._suggestions) {
+            obj[k] = v;
         }
-        fs.writeFileSync(ActBotConstants.suggestionsPath, fileString);
+        let temp = JSON.stringify(obj);
+        // console.log(temp);
+        return obj;
     }
 
     /**
      * Gets all suggestions in the list.
      */
-    public static getsuggestions(): string | undefined {
-        let listSize: number = SuggestionsManager._suggestions.length;
-        if (listSize > 0) {
-            let stringToReturn: string = "";
-            for (let x: number = 0; x < listSize; x++) {
-                stringToReturn = stringToReturn + SuggestionsManager._suggestions[x] + "\n";
-            }
-            return stringToReturn;
-
+    public static getsuggestions(list: string): string | undefined {
+        let templist = list.toLowerCase().trim();
+        if (SuggestionsManager._suggestions.has(templist)) {
+            let suggestionsToReturn = SuggestionsManager._suggestions.get(templist);
+            if (suggestionsToReturn!.toString() == "")
+                return "List is empty. Add a suggestion!";
+            return suggestionsToReturn!.toString().replace(/,/g, "\n");
         }
         return undefined;
     }
 
+    /**
+     * Gets all lists.
+     */
+    public static getlists(): string | undefined {
+        let tempLists = SuggestionsManager._suggestions.keyArray();
+        // console.log(tempLists.toString()).replace(/,/g, "\n");
+        return tempLists.toString().replace(/,/g, "\n");
+    }
+
     /**
      * Reads in the suggestions file from memory.
      */
     private static readsuggestionsFile() {
 
-        SuggestionsManager._suggestions = [];
+        SuggestionsManager._suggestions.clear();
+        let tempString = fs.readFileSync(ActBotConstants.suggestionsPath).toString("utf-8");
+        let tempParsed = JSON.parse(tempString) as Array<SuggestionLists>;
 
-        fs.readFile(ActBotConstants.suggestionsPath, (error, data) => {
-            let suggestions = data.toString().split(delimiter);
-            for (let i = 0; i < suggestions.length; i++) {
-                SuggestionsManager._suggestions[i] = suggestions[i];
-            }
-        });
+        for (let property in tempParsed) {
+            // console.log(property);
+            // console.log(tempParsed[property]);
+            if (!SuggestionsManager._suggestions.has(property.toLowerCase()))
+                SuggestionsManager._suggestions.set(property.toLowerCase(), tempParsed[property]);
+        }
+        let x = SuggestionsManager.collectionToObject();
+        console.log(x);
```

Comment:
Probably want to remove the debug log stuff

Reply:
At least where there's no particular context and it looks like it was being used to troubleshoot the code 😆 

We probably want to add a more robust logging system at some point but that will come later